apps/chat/src/store/conversations/conversations.epics.ts

```diff
@@ -2881,6 +2886,133 @@ const deleteChosenConversationsEpic: AppEpic = (action$, state$) =>
+        shouldUpload: payload.targetConversationId
+          ? of(
+              ConversationsSelectors.selectConversation(
+                state$.value,
+                payload.targetConversationId,
+              ),
+            ).pipe(map((conv) => !conv))
+          : of(false),
+      }),
+    ),
+    concatMap(
+      ({
+        conversation,
+        selectedModelId,
+        modelsMap,
+        addonsMap,
+        installedModelIds,
+        shouldUpload,
+      }) => {
+        const modelToApply = modelsMap[selectedModelId];
+
+        if (!modelToApply)
+          return of(
+            MarketplaceActions.setApplyModelStatus(CompletionStatus.FAILED),
+          );
+
+        return concat(
+          of(MarketplaceActions.setApplyModelStatus(CompletionStatus.STARTED)),
+          iif(
+            () => shouldUpload && !!conversation,
+            of(
+              ConversationsActions.addConversations({
+                conversations: [conversation as Conversation],
+              }),
+            ),
+            EMPTY,
+          ),
+          iif(
+            () => !!conversation,
+            of(
+              ConversationsActions.updateConversation({
+                id: conversation?.id as string,
+                values: {
+                  ...(conversation
+                    ? getConversationModelParams(
+                        conversation as Conversation,
+                        modelToApply?.reference,
+                        modelsMap,
+                        addonsMap,
+                      )
+                    : {}),
+                },
+              }),
+            ),
+            of(
+              ConversationsActions.createNewConversations({
+                names: [DEFAULT_CONVERSATION_NAME],
+                modelReference: modelToApply?.reference,
+              }),
+            ),
+          ),
+          of(
+            ModelsActions.updateRecentModels({
+              modelId: modelToApply?.reference as string,
+              rearrange: true,
+            }),
+          ),
+          iif(
+            () => !installedModelIds.has(modelToApply?.reference as string),
+            of(
+              ModelsActions.addInstalledModels({
+                references: [modelToApply?.reference as string],
+              }),
+            ),
+            EMPTY,
+          ),
+        );
+      },
+    ),
+    catchError(() =>
+      of(MarketplaceActions.setApplyModelStatus(CompletionStatus.FAILED)),
+    ),
+  );
+
+const applyMarketplaceModelSuccess: AppEpic = (action$, state$) =>
+  action$.pipe(
+    map((action) => ({
+      action,
+      status: MarketplaceSelectors.selectApplyModelStatus(state$.value),
+    })),
+    filter(
+      ({ action, status }) =>
+        (ConversationsActions.saveConversationSuccess.match(action) ||
+          (ConversationsActions.setIsActiveConversationRequest.match(action) &&
+            !action.payload)) &&
+        status === CompletionStatus.STARTED,
+    ),
```

Comment:
```suggestion
    filter((action) => {
      const status = MarketplaceSelectors.selectApplyModelStatus(state$.value);

      return (
        (ConversationsActions.saveConversationSuccess.match(action) ||
          (ConversationsActions.setIsActiveConversationRequest.match(action) &&
            !action.payload)) &&
        status === CompletionStatus.STARTED
      );
    }),
```

Reply:
is there any cost differences between these two implementations? If not, seems like first one is more declarative